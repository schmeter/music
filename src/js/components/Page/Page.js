import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Helmet } from 'react-helmet';

import ScrollTop from '../ScrollTop';
import { getTitle } from '../../services/meta';

class Page extends React.Component {
    render() {
        const { className, id, title, children, useBaseClass } = this.props;

        return (
            <>
                <Helmet>
                    <title>{title || getTitle(id)}</title>
                </Helmet>
                <ScrollTop target="main">
                    <main className="main">
                        <div className={classNames(
                            'page',
                            { 'page-base': useBaseClass !== false },
                            `page-${className || id}`
                        )}>
                            {children}
                        </div>
                        <audio
                            className="spacer"
                            controls
                        />
                    </main>
                </ScrollTop>
            </>
        );
    }
}

Page.propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string,
    className: PropTypes.string,
    children: PropTypes.node,
    useBaseClass: PropTypes.bool
};

export default Page;
