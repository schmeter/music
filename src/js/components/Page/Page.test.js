import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Page from './Page';

it('renders correctly with all parameters', () => {
    const props = {
        id: 'test',
        className: 'test',
        hasDynamicTitle: true,
        useBaseClass: false
    };

    const component = shallow(<Page {...props} >test</Page>);
    expect(toJson(component)).toMatchSnapshot();
});

it('renders correctly with id only', () => {
    const props = {
        id: 'test'
    };

    const component = shallow(<Page {...props} >test</Page>);
    expect(toJson(component)).toMatchSnapshot();
});
