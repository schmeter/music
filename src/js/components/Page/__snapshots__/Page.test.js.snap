// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all parameters 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`renders correctly with id only 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
